cli/medperf/entities/dataset.py

```diff
@@ -86,48 +94,8 @@ def is_ready(self):
         flag_file = os.path.join(self.path, config.ready_flag_file)
         return os.path.exists(flag_file)
 
-    def todict(self):
-        return self.extended_dict()
-
-    @classmethod
-    def all(cls, local_only: bool = False, filters: dict = {}) -> List["Dataset"]:
-        """Gets and creates instances of all the locally prepared datasets
-
-        Args:
-            local_only (bool, optional): Wether to retrieve only local entities. Defaults to False.
-            filters (dict, optional): key-value pairs specifying filters to apply to the list of entities.
-
-        Returns:
-            List[Dataset]: a list of Dataset instances.
-        """
-        logging.info("Retrieving all datasets")
-        dsets = []
-        if not local_only:
-            dsets = cls.__remote_all(filters=filters)
-
-        remote_uids = set([dset.id for dset in dsets])
-
-        local_dsets = cls.__local_all()
-
-        dsets += [dset for dset in local_dsets if dset.id not in remote_uids]
-
-        return dsets
-
-    @classmethod
-    def __remote_all(cls, filters: dict) -> List["Dataset"]:
-        dsets = []
-        try:
-            comms_fn = cls.__remote_prefilter(filters)
-            dsets_meta = comms_fn()
-            dsets = [cls(**meta) for meta in dsets_meta]
-        except CommunicationRetrievalError:
-            msg = "Couldn't retrieve all datasets from the server"
-            logging.warning(msg)
-
-        return dsets
-
     @classmethod
-    def __remote_prefilter(cls, filters: dict) -> callable:
+    def _Entity__remote_prefilter(cls, filters: dict) -> callable:
```

Comment:
This staff looks like a magic command. How is it called? in `interface` I still see `__remote_prefilter` is called
https://github.com/hasan7n/medperf/blob/refactor-entities/cli/medperf/entities/interface.py#L68

Reply:
yeah it's because we are overriding a superclass method that is private. I am thinking of making it public to avoid this kind of magic/weird things

Reply:
Can we make it "usual" private via naming `_remote_prefilter`? AFAIU in that case the method is visible to the children (i.e. can be overridden easily), but is kinda "hidden" from external code (i.e. if external code would call this method, any IDE would highlight that method is protected and you should not refer to it directly)

Reply:
Anyway, I agree, if we cannot use usual private naming, it's worth to make it public: right now IDE even does not show to me where the method is overridden (as it happens only dynamically during code execution)

Reply:
if we name it `_remote_prefilter`, it will not really override the superclass's method because it's private (tested it then figured out that it should be named in this weird way so that it works)

So yeah, I am gonna make it public for ease of understanding the code and for better editors intellisense

Reply:
I mean, call it `__remote_prefilter` + `@staticmethod` instead of `@classmethod`. We don't work with classes here really, and static methods are simpler to override